### PR TITLE
Adds ide plugin

### PR DESCRIPTION
**Pull request by i-walker**
This subproject can't access descriptors of the ide part of Meta, because there is no published version users can add in:
```groovy
// build.gradle
intellij {
    version '2019.3.3'
    pluginName = "My Ide Plugin"
    plugins = ["io.arrow-kt.arrow"]
}
``` 
thus the following doesn't work either:
```xml
<!--plugin.xml-->
<idea-plugin>
  <!--.. other dependencies-->
 <depends>io.arrow-kt.arrow</depends>
</idea-plugin>
```
Both of the aformentioned changes are remaining to create a hello world plugin.

### CODE REVIEW

**Review by rachelcarmena — CHANGES_REQUESTED** [review]
Thanks @i-walker :tada: Please, it's better if it's outside `hello-world` directory because that directory has 2 related projects. For instance, create `hello-world-ide-plugin`.

And add these lines:

```
        cd ../hello-world-ide-plugin
        ./gradlew clean build
```

at the end of `.github/workflows/build.yml`.

We'll improve it to make it easier afterwards :wink: 

Thanks!!

**Comment by rachelcarmena** [comment]
> This subproject can't access descriptors of the ide part of Meta, because there is no published version users can add in:
> 
> ```groovy
> // build.gradle
> intellij {
>     version '2019.3.3'
>     pluginName = "My Ide Plugin"
>     plugins = ["io.arrow-kt.arrow"]
> }
> ```
> 
> thus the following doesn't work either:
> 
> ```
> 
> <idea-plugin>
>   
>  <depends>io.arrow-kt.arrow</depends>
> </idea-plugin>
> ```
> 
> Both of the aformentioned changes are remaining to create a hello world plugin.

We still have a snapshot version in Arrow Meta so we're using a snapshot repository: https://meta.arrow-kt.io/idea-plugin/snapshots/1.3.61-SNAPSHOT/updatePlugins.xml

As soon as we have a release version, we can publish the plugins on official places: Jetbrains Plugin Repository in this case.

**Comment by i-walker** [comment]
Agree. But I would not merge this until we have an EAP or release version, because folks would expect an Ide example using Meta.
Maybe we can publish an experimental EAP of some sort.
What do you think?

**Comment by rachelcarmena** [comment]
> Agree. But I would not merge this until we have an EAP or release version, because folks would expect an Ide example using Meta.
> Maybe we can publish an experimental EAP of some sort.
> What do you think?

It can be an option, however, I don't think if I understand the purpose of this plugin and the difference with the IDEA plugin that can be installed now...

**Comment by i-walker** [comment]
The ide plugin that is proposed here is the one developers use to define their own Meta ide plugins, with the DSL. 
They depend on the one, which is in Arrow Meta.

**Comment by i-walker** [comment]
Unfortunately, a published snapshot is not enough for the IntelliJ Platform, to include the dependency in their classpath.
https://www.jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_dependencies.html

**Comment by rachelcarmena** [comment]
> The ide plugin that is proposed here is the one developers use to define their own Meta ide plugins, with the DSL.
> They depend on the one, which is in Arrow Meta.

The Hello World example for the Compiler Plugin shows how to create another compiler plugin with the Hello World plugin.

I think I was expecting the same here: how to create an IDEA plugin from Arrow IDEA plugin + Hello World Plugin. 

However, I've seen a difference: Compiler Plugin doesn't include Hello World plugin and IDEA Plugin already includes Hello World plugin. What if Hello World plugin is removed from IDEA Plugin and then is included here to show how it works?

**Comment by i-walker** [comment]
That's an option, we can explore. Though the same issue remains, which was raised here https://kotlinlang.slack.com/archives/CJ699L62W/p1581883669083900.

**Comment by Redrield** [comment]
Is there any timeline for when it'll be possible to make arrow IDE plugins?

**Comment by i-walker** [comment]
Excuse our late response @Redrield . 
We're setting up a 47 account for a Plugin registry to publish the library, so that we can resolve the dependency error's we currently have, for plugin developer depending on Meta. 

We will announce this in a public channel once that is done. 
From there I'll be adding a few examples and Docs.

File: hello-world/compiler-plugin/settings.gradle
Comment (rachelcarmena) [inline hello-world/compiler-plugin/settings.gradle]:
```suggestion
```

**Review by rachelcarmena — CHANGES_REQUESTED** [review]

**Comment by rachelcarmena** [comment]
@i-walker , mate, [this commit](https://github.com/arrow-kt/arrow-meta-examples/pull/19/commits/61c85a8cb64f113c8eb96cd096fbaa60b5b703ce) is removing necessary files. They are independent projects. On the other hand, `SHADOW_JAR_VERSION=5.2.0ide-plugin` could be removed because it's not being used.

**Comment by i-walker** [comment]
Yes, they should be dependent so that users can publish either their compiler-plugin separately or the ide-plugin with the compiler plugin.

**Comment by rachelcarmena** [comment]
> Yes, they should be dependent so that users can publish either their compiler-plugin separately or the ide-plugin with the compiler plugin.

The Idea plugin already imports the Compiler Plugin.

**Comment by i-walker** [comment]
I meant the compiler plugin the user creates not the one from Arrow.

File: hello-world/cli-and-ide-plugin/ide-plugin/build.gradle
Comment (rachelcarmena) [inline hello-world/cli-and-ide-plugin/ide-plugin/build.gradle:36]: Despite of having the IDE plugin "pending of approval" in Jetbrains repository, it can be resolved.

So `"io.arrow-kt.arrow:${ARROW_META_VERSION}"` refers to the plugin which was published 2 weeks ago.

**How to verify it?**

- Remove the compiler-plugin in `settings.gradle`
- Remove the dependencies in `ide-plugin/build.gradle`
- Remove OSS repository in `build.gradle`

The build check will continue working, because `"io.arrow-kt.arrow:${ARROW_META_VERSION}"` is being found in Jetbrains repository instead of OSS.

**Another check to verify it**, look at the content of the sandbox:

```
ide-plugin/build/idea-sandbox/plugins
├── Arrow Meta Intellij IDEA Plugin
│   └── lib
├── Example
│   └── lib
└── Kotlin
    ├── bin
    ├── kotlinc
    └── lib
```

The Arrow Meta plugin is called `Arrow Meta Intellij IDEA Plugin` and that was changed by `Arrow Meta` a few days ago.

File: hello-world/cli-and-ide-plugin/build.gradle
Comment (rachelcarmena) [inline hello-world/cli-and-ide-plugin/build.gradle:13]: It's not necessary to specify this repository because it's the repository used by default for the Intellij Gradle Plugin.

**Review by rachelcarmena — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
